sysconfig/sysconfig.go

```diff
@@ -58,13 +58,19 @@ type Options struct {
 type FilesystemOnlyApplyOptions struct {
 	// Classic is true when the system in rootdir is a classic system
 	Classic bool
+	// UC20 is true when the system in rootdir is a UC20 system
```

Comment:
we are either pointing to a root filesystem  or to the recovery partition, no, which are vatly different? so it's a bit unclear the flag here is useful

Reply:
I think some of your confusion here comes from the fact that while yes we have either a recovery system or a root filesystem, the code doesn't really matter much, the only choice is whether we use `<dir>/boot/uboot/config.txt` or `<dir>/config.txt` or `/run/mnt/ubuntu-seed/config.txt`. In the last case, we always know when to do that by consulting the kernel command line, and the first and second choices are indicated by whether the system being prepared is UC20 or not.

To simplify things here, I have limited the functionality here to just work on UC20 recovery systems, but made clear in the comments that we could relatively easily enable UC18 root filesystems at ubuntu-image/prepare-image time with this functionality _if_ we could decide on how to indicate that from the caller. Hopefully this all makes sense now.